examples/posenet/src/PoseNet.js

```diff
@@ -0,0 +1,372 @@
+class PoseNet{
+  constructor(tfmodel, backend, version, outputstride, input, type){
+    this._tfmodel = tfmodel;
+    this._model = null;
+    this._compilation;
+    this._execution;
+    this._tensorIds = [];
+    this._operandIndex = 0;
+    this._version = version;
+    this._outputstride = outputstride;
+    this._inputs = input;
+    this._shape = [];
+    this._outputs = [];
+    this._type = type;
+    if (typeof backend !== 'undefined') {
+      this._backend = backend;
+    } 
+    else {
+      if (nnNative) {
+        this._backend = 'WebML';
+      } 
+      else {
+        this._backend = 'WASM';
+      }
+    }
+    if (this._backend === 'WebML') {
+      if (nnNative === null) {
+        throw Error('Fails to initialize neural network context');
+      }
+      this._nn = nnNative;
+    }
+    else if (this._backend === 'WASM' || this._backend === 'WebGL2') {
+      this._nn = nnPolyfill;
+    }
+  }
+  async createCompiledModel(){
+    let options = {};
+    if(this._backend === 'WebGL2'){
+      options.useWebGL2 = true;
+    }
+    this._model = await this._nn.createModel(options);
+    
+    await this._addTensorOperands();
+    this._addOpsAndParams();
+    
+    await this._model.finish();
+    this._compilation = await this._model.createCompilation();
+    this._compilation.setPreference(this._nn.PREFER_FAST_SINGLE_ANSWER);
+    await this._compilation.finish();
+    this._execution = await this._compilation.createExecution();
+  }
+
+  async compute_single(inputTensor, heatmapTensor, offsetTensor){
+    this._execution.setInput(0, inputTensor);
+    this._execution.setOutput(0, heatmapTensor);
+    this._execution.setOutput(1, offsetTensor);
+    let error = await this._execution.startCompute();
+    if(error){
+      return error;
+    }
+    return 'success';
+  }
+
+  async compute_multi(inputTensor, heatmapTensor, offsetTensor, displacement_fwd, displacement_bwd){
+    this._execution.setInput(0, inputTensor);
+    this._execution.setOutput(0, heatmapTensor);
+    this._execution.setOutput(1, offsetTensor);
+    this._execution.setOutput(2, displacement_fwd);
+    this._execution.setOutput(3, displacement_bwd);
+    let error = await this._execution.startCompute();
+    if(error){
+      return error;
+    }
+    return 'success';
+  }
+
+
+  async _addTensorOperands(){
+    this._tfmodel = toOutputStridedLayers(this._tfmodel, this._outputstride);
+    let dimension_out;
+    let dimension_in = this._inputs;
+    const type = this._nn.TENSOR_FLOAT32;
+    for(let i in this._tfmodel){
+      let dimension = [];
+      let weights = [];
+      let dimension_bias = [];
+      let bias = [];
+      if(this._tfmodel[i]["convType"] === "conv2d"){
+        await getDimensionData("conv2d", this._version, i).then(function(data){
```

Comment:
When use `await`, you don't need to use `then`. You can get the result by `let data = await getDimensionData(...)`.